### ISSUE TITLE
Legend alignment in dotplot

### ISSUE THREAD

**Issue by LuckyMD** [scverse/scanpy#1103]
Hey!
The color bar and legend don't seem to align well... might be because i'm subsetting the object.

Here is what happens (reproducible example and my plot, which is even stranger with legend overlap):

<img width="623" alt="Screenshot 2020-03-15 at 13 09 33" src="https://user-images.githubusercontent.com/13019956/76701118-73bff200-66be-11ea-9c99-ee4e0427bbfa.png">
<img width="580" alt="Screenshot 2020-03-15 at 13 02 52" src="https://user-images.githubusercontent.com/13019956/76701119-76224c00-66be-11ea-9b45-b94c1d9a57bd.png">

```python
import scanpy as sc
adata = sc.datasets.pbmc68k_reduced()
sc.pl.dotplot(adata[adata.obs.bulk_labels.isin(['Dendritic', 'CD14+ Monocyte', 'CD4+/CD25 T Reg']),], var_names=['HES4', 'TNFRSF4', 'SSU72'], groupby='bulk_labels', figsize=(8,8))

```

#### Versions:

> scanpy==1.4.5.1 anndata==0.7.1 umap==0.3.10 numpy==1.18.1 scipy==1.4.1 pandas==1.0.1 scikit-learn==0.22.2 statsmodels==0.11.1 python-igraph==0.8.0 louvain==0.6.1

Also, matplotlib is 3.1.3

**Comment by ivirshup** [comment]
Huh. As far as I can tell, this seems to happen only if there are 3 or less groups.

**Comment by ivirshup** [comment]
Probably has something to do with this:

https://github.com/theislab/scanpy/blob/51dc038cb73fc1798373e33d990090439ec57e9e/scanpy/plotting/_anndata.py#L1896-L1905

**Comment by LuckyMD** [comment]
This seems like a bad idea, no? It also seems to throw off the legend other than the colorbar.

**Comment by fidelram** [comment]
In #1127 I changed the way that the colorbar and size legend are plotted and should solve this problem.

**Comment by gabriellayi** [comment]
Hi, I have the same problem, sorry that I just started learning python, so I don't really understand some of the improvements and I also tried some of them, it didn't work. 
Could you please let me know what kind of solution will be good for this issue?

Thanks! 
<img width="765" alt="Screen Shot 2020-05-24 at 17 43 08" src="https://user-images.githubusercontent.com/50899584/82768801-15ae3a00-9de6-11ea-9552-88eb59b19405.png">
<img width="324" alt="Screen Shot 2020-05-24 at 17 43 42" src="https://user-images.githubusercontent.com/50899584/82768808-2199fc00-9de6-11ea-8624-61bd67c5aae7.png">

Yi

**Comment by fidelram** [comment]
This is being fixed in #1210

for the time being you can try to set `dendrogram=False`  or you can install the branch with the fix.
